modules/home-environment.nix

```diff
@@ -555,11 +577,17 @@ in
         export __HM_SESS_VARS_SOURCED=1
 
         ${config.lib.shell.exportAll cfg.sessionVariables}
-      '' + lib.optionalString (cfg.sessionPath != [ ]) ''
-        export PATH="$PATH''${PATH:+:}${concatStringsSep ":" cfg.sessionPath}"
-      '' + cfg.sessionVariablesExtra;
+      '' + lib.concatStringsSep "\n"
+        (lib.mapAttrsToList
+          (env: values: ''
+            export ${env}="''$${env}''${${env}:+:}${lib.concatStringsSep ":" values}"'')
+          cfg.sessionSearchVariables)
+        + cfg.sessionVariablesExtra;
     };
 
+    home.sessionSearchVariables.PATH =
+      lib.optionals (cfg.sessionPath != [ ]) cfg.sessionPath;
```

Comment:
Tested by this and it is still passing locally: https://github.com/nix-community/home-manager/pull/4582/files#diff-cdf6113d7d7a7dd00dc882684b388b8a0e537bb6488d28c7e736a8eccf6b9186R2